stripe.go

```diff
@@ -571,18 +571,35 @@ func (s *BackendImplementation) UnmarshalJSONVerbose(statusCode int, body []byte
 // Checks if an error is a problem that we should retry on. This includes both
 // socket errors that may represent an intermittent problem and some special
 // HTTP statuses.
-func (s *BackendImplementation) shouldRetry(err error, resp *http.Response, numRetries int) bool {
+func (s *BackendImplementation) shouldRetry(err error, req *http.Request, resp *http.Response, numRetries int) bool {
 	if numRetries >= s.MaxNetworkRetries {
 		return false
 	}
 
+	// TODO: There are many errors that should not be retried. Try to make this
+	// more granular by including only connection errors, timeout errors, etc.
```

Comment:
Added this TODO here because honestly what we're doing right now isn't great.

Unfortunately, routing around Go source looking for errors that we can compare against is slightly non-trivial, so deferring this to a separate improvement project.